wire-ios-data-model/Source/Proteus/ProteusProvider.swift

```diff
@@ -22,26 +22,42 @@ import WireUtilities
 public typealias KeyStorePerformBlock<T> = ((UserClientKeysStore) throws -> T)
 public typealias ProteusServicePerformBlock<T> = ((ProteusServiceInterface) throws -> T)
 
+public typealias KeyStorePerformAsyncBlock<T> = ((UserClientKeysStore) async throws -> T)
+public typealias ProteusServicePerformAsyncBlock<T> = ((ProteusServiceInterface) async throws -> T)
+
 public protocol ProteusProviding {
 
     func perform<T>(
         withProteusService proteusServiceBlock: ProteusServicePerformBlock<T>,
         withKeyStore keyStoreBlock: KeyStorePerformBlock<T>
     ) rethrows -> T
 
+    func performAsync<T>(
+        withProteusService proteusServiceBlock: ProteusServicePerformAsyncBlock<T>,
+        withKeyStore keyStoreBlock: KeyStorePerformAsyncBlock<T>
+    ) async rethrows -> T
+
     var canPerform: Bool { get }
 }
 
 public class ProteusProvider: ProteusProviding {
 
-    private let context: NSManagedObjectContext
+    private let proteusService: ProteusServiceInterface?
+    private let keyStore: UserClientKeysStore
     private let proteusViaCoreCrypto: Bool
 
-    public init(
+    public convenience init(
```

Comment:
`ProteusProvider` could be a struct, right?

Reply:
yes, it could be, do you see issues with the class now?

Reply:
No, I don't.